docs/source/user_guide/data_checks.ipynb

```diff
@@ -54,7 +54,150 @@
+    "* `NoVarianceDataCheck`\n",
+    "* `HighVarianceCVDataCheck`\n",
+    "* `InvalidTargetDataCheck`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Zero Variance\n",
+    "\n",
+    "Data with zero variance indicates that all values are identical. If a feature has zero variance, it is not likely to be a useful feature. Similarly, if the target has zero variance, there is likely something wrong. `NoVarianceDataCheck` checks if the target or any feature has only one unique value and alerts you to any such columns."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import NoVarianceDataCheck\n",
+    "X = pd.DataFrame({\"no var col\": [0, 0, 0],\n",
+    "                 \"good col\":[0, 4, 1]})\n",
+    "y = pd.Series([1, 0, 1])\n",
+    "no_variance_data_check = NoVarianceDataCheck()\n",
+    "\n",
+    "for message in no_variance_data_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that `NaN` counts as an unique value, but `NoVarianceDataCheck` will still return a warning if there is only one unique non-`NaN` value in a given column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import NoVarianceDataCheck\n",
+    "\n",
+    "X = pd.DataFrame({\"no var col\": [0, 0, 0],\n",
+    "                 \"no var col with nan\": [1, np.nan, 1],\n",
+    "                 \"good col\":[0, 4, 1]})\n",
+    "y = pd.Series([1, 0, 1])\n",
+    "\n",
+    "no_variance_data_check = NoVarianceDataCheck()\n",
+    "\n",
+    "for message in no_variance_data_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Class Imbalance\n",
+    "\n",
+    "For classification problems, the distribution of examples across each class can vary. For small variations, this is normal and expected. However, when the number of examples for each class label is disproportionately biased or skewed towards a particular class (or classes), it can be difficult for machine learning models to predict well. In addition, having a low number of examples for a given class could mean that one or more of the CV folds generated for the training data could only have few or no examples from that class. This may cause the model to only predict the majority class and ultimately resulting in a poor-performant model.\n",
+    "\n",
+    "`ClassImbalanceDataCheck` checks if the target labels are imbalanced beyond a specified threshold for a certain number of CV folds. It returns errors for any classes that have less samples than double the number of CV folds, and warnings for any classes that fall below the set threshold percentage."
```

Comment:
What are your thoughts on explaining the difference between a `DataCheckError` and `DataCheckWarning` ?Sounds intuitive to me but it might be confusing to a newcomer. Maybe instead of "returns errors .... and warnings" we should use `returns DataCheckErrors ...` so that they don't expect an exception ? 

I noticed this might be confusing because in the "Writing your own DataCheck" section, we import `DataCheckError` without introducing the concept beforehand.

Reply:
Good idea! Added a small section about `DataCheckMessage` classes and updated to reference those classes :D